woof

```diff
@@ -166,6 +166,12 @@ class FileServHTTPRequestHandler (http.server.BaseHTTPRequestHandler):
 
 
    def do_GET (self):
+      return self.get_or_head ()
+
+   def do_HEAD (self):
+      return self.get_or_head (True)
+
+   def get_or_head (self, headers_only=False):
```

Comment:
This is an anti-pattern to have a method with a flag true/false.

See for example https://martinfowler.com/bliki/FlagArgument.html and as indicated there it creates here a good example of tangled code.

What about two separate methods:
  - one to send headers
  - one to send contents

With that, do_GET() can call both of them, do_HEAD() will only call the headers one.

And yes, that would be more clean if a previous commit split in two different methods the upload form and the file to serve.

Reply:
Curses, I lost my reply. This patch does **exactly** what that article recommends when the logic is "messy". Here the "mess" is multiple return points, and the exposed APIs is still two separate methods, with `do_get_or_head` never intended to be called directly.

If you really think it needs to be broken up into two methods, propose a patch that does so. If going that route, there's no need for hidden methods, `do_GET` can call `do_HEAD` as its first operation.